Orange/widgets/unsupervised/owdistances.py

```diff
@@ -160,6 +161,8 @@ def checks(metric, data):
         except MemoryError:
             self.Error.distances_memory_error()
             return
+
+        met = np.nan_to_num(met)
```

Comment:
No. This is just a quick local patch, not a solution. 

It reminds me of a guy (whom we all know and) who - years ago - programmed some machine learning algorithm. He occasionally got probabilities above 1, so he fixed the problem by adding

    if (p > 1) {
        p = 1;
    }

If we get an error report, we shouldn't find the simplest way to just fix it somewhere. If some distances are infinite, they are infinite. We can't just set them to 42 and pretend everything is cool. If we believe that distance matrix should never contain infinite values, we should raise an exception. If we decide that infinite values can arise and are legal, other functions/widgets should be able to handle those.

In this case, I'd say that infinite values are allowed and hierarchical clustering should do something about them. Replacing them with large values (perhaps `len(data) * np.nanmax(data)`) could make sense within the hierarchical clustering. Or it may be even more complex since it depends upon linkage.

So: if there are infinite distances, we need to think what would we like the clustering to do with them, not what is the easiest way to get rid of them.